x-pack/plugins/reporting/server/lib/screenshots/observable.ts

```diff
@@ -123,6 +123,7 @@ export function screenshotsObservableFactory(
             );
 
             return setup$.pipe(
+              takeUntil(exit$),
```

Comment:
This immediately causes the job to fail if there is an error on the page that causes the process to exit. The same thing is already handled in the above code, but it also needs to go here, in the event this part of the process is where a crash happens.

Reply:
I went back to having this `takeUntil` being only once in the setup of the pipeline. Now, we do more "polling" of the browser driver's `Page` object to see if it's still open. That seems to be the best way to make sure the correct error is thrown.